interpreter.go

```diff
@@ -271,15 +477,17 @@ func makeCallStack(limit int) callStack {
 // TODO(dcunnin): Add string output.
 // TODO(dcunnin): Add multi output.
 
+// Keeps current execution context and evaluates things
 type interpreter struct {
-	stack          callStack
-	idArrayElement identifier
-	idInvariant    identifier
-	externalVars   vmExtMap
+	stack          callStack  // TODO what is it?
+	idArrayElement identifier // TODO what is it?
```

Comment:
Array elements are lazy, so I re-used the thunk logic for them.  However the thunk requires a variable name (it is used in error messages).  So these are placeholders to put in that slot when using the thunk for other purposes:  1) array elements 2) object invariants (asserts).  The text of the identifier is "array_element" and "object_assert" and appears in the following stacktraces:

```
$ jsonnet -e '[ error "foo" ]'
RUNTIME ERROR: foo
        <cmdline>:1:3-13        thunk <array_element>
        During manifestation
```

```
$ jsonnet -e '{ assert false }'
RUNTIME ERROR: Object assertion failed.
        <cmdline>:1:10-14       thunk <object_assert>
        During manifestation
```